src/homepage/_posts/2020-04-06-james-3.5.0.markdown

```diff
@@ -0,0 +1,83 @@
+---
+layout: post
+title:  "Apache James Server 3.5.0"
+date:   2020-04-06  15:16:30 +0200
+categories: james update
+---
+
+The Apache James developers are pleased to announce James server 3.5.0 release.
+
+Here are some points we worked on:
+
+### Added
+- Distributed task management for Guice cassandra-rabbitmq product. This enables several James servers to share a consistent view
+of tasks being currently executed.
+- JAMES-2563 Health check for ElasticSearch
+- JAMES-2904 Authentication and SSL support for Cassandra backend
+- JAMES-2904 Authentication and SSL support for ElasticSearch backend
+- JAMES-3066 Add "allowed From headers" webadmin endpoint
+- JAMES-3062 EventDeadLettersHealthCheck
+- JAMES-3058 WebAdmin offline task to correct mailbox inconsistencies on top of Cassandra products
+- JAMES-3105 WebAdmin offline task to recompute mailbox counters on top of Cassandra products
+- JAMES-3072 Webadmin endpoint to export mailbox backup
+
+### Changed
+- Multiple changes have been made to enhance ElasticSearch performance:
```

Comment:
why some entries have JIRA numbers but not all?

Reply:
Because people filled badly the changelog.

Reply:
could we fix it now?

Reply:
I will have a try but that could be very consumming.

Reply:
Maintaining a changelog is something that has been on my mind for some time.

How do you do it in James now?

My current view on this is is :
- Use a tool to extract the changelog from commits. Something like https://github.com/github-changelog-generator/github-changelog-generator ? I'm sure there are others.
- Manually draft the release notes from the generated changelog and publish them alongside the changelog. I've seen this in PostgreSQL - they do a release once a year and they start the release notes when they do the first RC (about now) with release in late October.